### PR TITLE
CS-18 Using push-state

### PR DESCRIPTION
**Pull request by steodor**
Handling client paths on server and delivering index.html accordingly; Updated routes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "lodash": "^4.0.0",
     "mocha": "^2.4.5",
     "nock": "^7.2.2",
+    "node-jsx": "^0.13.3",
```
Comment (ionutmandra) [inline package.json]: Check if needed

File: server.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // nodemon server.js
 
 //node modules
+require("node-jsx").install();
```
Comment (ionutmandra) [inline server.js]: check if required

File: server/routesCustom.js [hunk 3]
```diff
@@ -96,15 +103,43 @@ function setAdminRoutes(router){
 		});
 };
 
+function catch404(router){
```
Comment (ionutmandra) [inline server/routesCustom.js]: Create task to check if refactoring possible

**Comment by steodor** [comment]
Ok, `node-jsx` was removed. Task was created in Jira (CS-27)

**Comment by steodor** [comment]
I also refactored the `catch404()` function to use the `path-to-regexp` module which is what the `express` router uses itself for matching routes, effectively reducing it to a single, more simple middleware function.

**Pull request merged**